external_compressors/zstd_external.py

```diff
@@ -0,0 +1,170 @@
+"""External compressors zstd for testing/benchmarking purposes.
+
+Zstd is a modern LZ77 based compressor, using ANS for encoding match lengths and distances, and
+Huffman coding for literals.
+
+Zstd format is described in https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md.
+
+Zstd website: https://github.com/facebook/zstd/.
+
+We use the python zstandard module which internally calls the C library.
+"""
+
+import os
+import tempfile
+from core.data_block import DataBlock
+from core.data_encoder_decoder import DataDecoder, DataEncoder
+from core.data_stream import TextFileDataStream, Uint8FileDataStream
+from core.encoded_stream import EncodedBlockReader, EncodedBlockWriter
+from core.prob_dist import ProbabilityDist
+from utils.bitarray_utils import BitArray, bitarray_to_uint, uint_to_bitarray
+from utils.test_utils import (
+    create_random_binary_file,
+    try_file_lossless_compression,
+    try_lossless_compression,
+)
+import zstandard
+
+
+def create_zstd_compressor_object(level: int):
```

Comment:
maybe add these to individual classes as static methods? (as we will never use them independently?..